src/main/java/core/basesyntax/service/impl/WriterImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.Writer;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collections;
+
+public class WriterImpl implements Writer {
+    @Override
+    public File write(String content, String filePath) {
+        if (content == null) {
+            throw new RuntimeException("Content cannot be null, you should write something");
+        }
+        File output = new File(filePath);
+        try {
+            Files.write(output.toPath(), Collections.singleton(content));
```

Comment:
again: what does Collections.singleton() mean?
this is not a mistake, but I need you to answer this question

Reply:
I know that singleton is a pattern , but it seems to me that it is not about pattern here . 
Actually I don`t know exactly why we need to use Singleton here , maybe because we need something which is Iterable and Collection.singleton() gives us Immutable set with only one object and as we know set is Iterable.